scalr/datadog_checks/scalr/check.py

```diff
@@ -0,0 +1,128 @@
+        "runs-count": "runs.count",
+        "runs-successful": "runs.successful",
+        "runs-failed": "runs.failed",
+        "runs-awaiting-confirmation": "runs.awaiting_confirmation",
+        "runs-concurrency": "runs.concurrency",
+        "runs-queue-size": "runs.queue_size",
+        "quota-max-concurrency": "quota.max_concurrency",
+        "billings-runs-count": "billing.runs.count",
+        "billings-run-minutes-count": "billing.run_minutes.count",
+        "billings-flex-runs-count": "billing.flex_runs.count",
+        "billings-flex-runs-minutes-count": "billing.flex_run_minutes.count",
+    }
+
+    def __init__(self, name, init_config, instances):
+        super(ScalrCheck, self).__init__(name, init_config, instances)
+
+        self._validate_instance(self.instance)
+        self.url = self.instance.get(SCALR_URL_PARAM)
+        self.token = self.instance.get(SCALR_ACCESS_TOKEN_PARAM)
+        self.account_id = self._get_account_id()
+
+    def check(self, instance):
+
+        try:
+            response_json = self._get_json(SCALR_DD_METRICS_ENDPOINT.format(self.url, self.account_id))
+
+            for key, name in self.SCALR_ACCOUNT_METRICS.items():
+                if key in response_json and response_json[key] is not None:
+                    self.gauge(name, response_json[key], tags=instance.get('tags', []))
+
+        except Timeout as e:
+            self.service_check(
+                self.SERVICE_CHECK_NAME,
+                AgentCheck.CRITICAL,
+                message="Request timeout: {}, {}".format(self.url, e),
+            )
+            self.log.exception("Communication with Scalr timed out. %s", e)
+
+        except (HTTPError, InvalidURL, ConnectionError) as e:
+            self.service_check(
+                self.SERVICE_CHECK_NAME,
+                AgentCheck.CRITICAL,
+                message="Request failed: {}, {}".format(self.url, e),
+            )
+            self.log.exception("Couldn't reach Scalr. %s", e)
+
+        except JSONDecodeError as e:
+            self.service_check(
+                self.SERVICE_CHECK_NAME,
+                AgentCheck.CRITICAL,
+                message="JSON Parse failed: {}, {}".format(self.url, e),
+            )
+            self.log.exception("Unexpected response from Scalr. %s", e)
+
+        except ValueError as e:
+            self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL, message=str(e))
+            self.log.exception(str(e))
+
+    @staticmethod
+    def _validate_instance(instance):
+        """
+        Validate that all required parameters are present in the instance.
+        """
+        missing = []
+        for option in (SCALR_URL_PARAM, SCALR_ACCESS_TOKEN_PARAM):
+            if option not in instance:
+                missing.append(option)
+
+        if missing:
+            missing = ", ".join(missing)
+            raise errors.ConfigMissingError("Scalr instance configuration missing option(s): {}".format(missing))
```

Comment:
This will never be the case because you declared those parameters `required: true` in the `spec.yaml` file so the base check will automatically validate them for you. You can remove this method